src/scripts/main.js

```diff
@@ -1,3 +1,390 @@
 'use strict';
 
-// write your code here
+const start = document.querySelector('.start');
+const score = document.querySelector('.game-score');
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+score.textContent = 0;
+
+const cells = {
+  1: '',
+  2: '',
+  3: '',
+  4: '',
+  5: '',
+  6: '',
+  7: '',
+  8: '',
+  9: '',
+  10: '',
+  11: '',
+  12: '',
+  13: '',
+  14: '',
+  15: '',
+  16: '',
+};
+
+const colorMods = [
+  'field-cell--2',
+  'field-cell--4',
+  'field-cell--8',
+  'field-cell--16',
+  'field-cell--32',
+  'field-cell--64',
+  'field-cell--128',
+  'field-cell--256',
+  'field-cell--512',
+  'field-cell--1024',
+  'field-cell--2048',
+];
+
+function getRandom(min, max) {
+  return Math.round((min - 0.5) + Math.random() * (max - min + 1));
+}
+
+function removeColors() {
+  for (const key in cells) {
+    for (const mod of colorMods) {
+      if (document.querySelector(`.cell_${key}`).classList.contains(mod)) {
+        document.querySelector(`.cell_${key}`).classList.remove(mod);
+      }
+    }
+  }
+}
+
+function sync() {
+  for (const key in cells) {
+    document.querySelector(`.cell_${key}`).textContent = cells[key];
+
+    switch (cells[key]) {
+      case 2:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--2');
+        break;
+      case 4:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--4');
+        break;
+      case 8:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--8');
+        break;
+      case 16:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--16');
+        break;
+      case 32:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--32');
+        break;
+      case 64:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--64');
+        break;
+      case 128:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--128');
+        break;
+      case 256:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--256');
+        break;
+      case 512:
+        document.querySelector(`.cell_${key}`).classList.add('field-cell--512');
+        break;
+      case 1024:
+        document.querySelector(`.cell_${key}`)
+          .classList.add('field-cell--1024');
+        break;
+      case 2048:
+        document.querySelector(`.cell_${key}`)
+          .classList.add('field-cell--2048');
+        break;
+    }
+  }
+}
+
+function addNumbers() {
+  let cellNum1 = getRandom(1, 16);
+  let cellNum2 = getRandom(1, 16);
+
+  while (cells[cellNum1] !== '') {
```

Comment:
```suggestion
  while (cells[cellNum1]) {
```
empty string is a [falsy ](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)value